src/continuum_absorption/absorption_ff_metal.jl

```diff
@@ -0,0 +1,458 @@
+- `ne`: the number density of free electrons.
+"""
+metal_ff_absorption(ν, T, Z, ni, ne) = hydrogenic_ff_absorption(ν, T, Z, ni, ne)
+
+
+
+"""
+    all_metal_ff_absorption(ν::Real, T::Real, number_densities::Dict, ne::Real,
+                            exclude_species_with_departure_terms::Bool = false)
+
+Computes the free-free linear absorption coefficient (in cm⁻¹) of metal species. This only includes
+free-free absorption for species with a net positive charge of 1, 2, or 3. It also skips Hydrogenic
+atoms.
+
+# Arguments
+- `ν`: frequency in Hz
+- `T`: temperature in K
+- `number_densities` is a `Dict` mapping each `Species` to its number density
+- `ne`: the number density of free electrons.
+
+This uses the hydrogenic approximation for all species except for those that have departure terms.
+
+!!! note
+    This mostly exists for demonstrative purposes.
+
+    It might be useful to adopt this function for long-term use for efficiency reasons (even though
+    it departs from the format of all other continuum absorption calculations).  
+
+"""
+function all_metal_ff_absorption(ν::Real, T::Real, number_densities::Dict, ne::Real,
+                                 exclude_species_with_departure_terms::Bool = false)
+
+    error("I think we may need the user to pass in the partition functions for every species")
+
+    # recall, name of the free-free interaction differs from the name of the participating species
+    departure_dict = Dict([(species"He_II", _He_I_ff),
+                           (species"C_II",  _C_I_ff),
+                           (species"C_III", _C_II_ff),
+                           (species"Si_II", _Si_I_ff),
+                           (species"Mg_II", _Mg_I_ff)])
+
+    ndens_Z1 = 0.0
+    ndens_Z2 = 0.0
+    ndens_Z3 = 0.0
+
+    α_out = 0.0 * ν
+
+    for (k,ndens) in number_densities
+        if k in [species"H_II", species"He_III", species"Li_IV"]
+            # Honestly, it probably only makes sense to treat Hydrogen separately from this
+            # function
+
+            # skip cases where the species has not electrons, and is perfectly described by the
+            # hydrogenic free-free absorption HI free-free, He II free-free, & Li III free-free
+            continue
+        elseif (exclude_species_with_departure_terms && (k in departure_dict))
+            continue
+        elseif k in departure_dict
+            α_out += departure_dict[k](ν, T, ndens, ne)
+        else
+            if (k.charge == 1)     # example: species"O_II"
+                ndens_Z1 += ndens
+            elseif (k.charge == 2) # example: species"O_III"
+                ndens_Z2 += ndens
+            elseif (k.charge == 3) # example: species"O_IV"
+                ndens_Z3 += ndens
+            end
+        end
+    end
+
+    α_out += metal_ff_absorption(ν, T, 1, ndens_Z1, ne)
+    α_out += metal_ff_absorption(ν, T, 2, ndens_Z2, ne)
+    α_out += metal_ff_absorption(ν, T, 1, ndens_Z3, ne)
+    α_out
+end
```

Comment:
I don't understand how this function works...  I will bother you about it in person.